es-core/src/components/ImageGridComponent.h

```diff
@@ -276,12 +308,188 @@ void ImageGridComponent<T>::onSizeChanged()
+			if (nextTargetRange.min > mTargetRange.max
+				|| nextTargetRange.max < mTargetRange.min)
+			{
+				std::cout << "+++++++++++++ EMERGENCY MOD ACTIVATED +++++++++++++" << std::endl;
+				mEmergencyMod = true;
+			}
+
+			mTargetRange = nextTargetRange;
+			mLastStartPos = startPos;
+		}
+	}
+
+	dynamicImageLoader();
+}
+
+template<typename T>
+void ImageGridComponent<T>::dynamicImageLoader()
+{
+	// Exit if not visible
+	if (!mVisible)
+		return;
+
+	bool tilesNeedUpdate = false;
+
+	// Load new range
+	for (int i = mTargetRange.min; i <= mTargetRange.max; i++)
+	{
+		if (loadTexture(i))
+			tilesNeedUpdate = true;
+	}
+
+	if (mEmergencyMod)
+	{
+		// Emergency mod got activated, we unload all current range
+		for (int i = mCurrentRange.min; i <= mCurrentRange.max; i++)
+			unloadTexture(i);
+
+		mEmergencyMod = false;
+		std::cout << "+++++++++++++ EMERGENCY MOD DESACTIVATED +++++++++++++" << std::endl;
+	}
+	else
+	{
+		if (mCurrentRange.min < mTargetRange.min)
+			std::cout << ">> Unloading " << mCurrentRange.min << " to " << mTargetRange.min - 1 << std::endl;
+
+		// Unload before new range
+		for (int i = mCurrentRange.min; i < mTargetRange.min; i++)
+		{
+			if (unloadTexture(i))
+				tilesNeedUpdate = true;
+		}
+
+		if (mTargetRange.max + 1 <= mCurrentRange.max)
+			std::cout << ">> Unloading " << mTargetRange.max + 1 << " to " << mCurrentRange.max << std::endl;
+
+		// Unload after new range
+		for (int i = mTargetRange.max + 1; i <= mCurrentRange.max; i++)
+		{
+			if (unloadTexture(i))
+				tilesNeedUpdate = true;
+		}
+	}
+
+	mCurrentRange.min = mTargetRange.min;
+	mCurrentRange.max = mTargetRange.max;
+
+	// Update the tiles only if either a call to unloadTexture or loadTexture was successful
+	if (tilesNeedUpdate)
+		updateTiles();
+}
+
+template<typename T>
+std::shared_ptr<TextureResource> ImageGridComponent<T>::getDefaultTexture(const T& obj)
+{
+	// FileType::FOLDER = 2, but FileData is our template parameter T,
+	// so we don't want to bring that dependence to FileData here
+	return obj->getType() == 2 ?
+		   mDefaultFolderTexture :
+		   mDefaultGameTexture;
+}
+
+// Will load an object's texture from it's stored texture path.
+// Can be safely called many times without reloading texture.
+template<typename T>
+bool ImageGridComponent<T>::loadTexture(int index)
+{
+	if (!mEntries[index].data.isTextureLoaded)
+	{
+		std::string imagePath = mEntries[index].data.texturePath;
+
+		if (ResourceManager::getInstance()->fileExists(imagePath))
+			mEntries[index].data.texture = TextureResource::get(imagePath);
+
+		mEntries[index].data.isTextureLoaded = true;
+
+		return true;
+	}
+
+	return false;
+}
+
+// Sets this image back to default texture.
+template<typename T>
+bool ImageGridComponent<T>::unloadTexture(int index)
+{
+	if (mEntries[index].data.isTextureLoaded)
+	{
+		mEntries[index].data.texture = getDefaultTexture(mEntries[index].object);
```

Comment:
In the sense that why do we even do this in the first place? Why not leave it with the actual texture? I wonder if this is not making the texture memory management a bit more constrained than it could.